pnet_datalink/src/lib.rs

```diff
@@ -181,8 +181,18 @@ impl Default for Config {
 /// When matching on the returned channel, make sure to include a catch-all so that code doesn't
 /// break when new channel types are added.
 #[inline]
-pub fn channel(network_interface: &NetworkInterface, configuration: Config) -> io::Result<Channel> {
-    backend::channel(network_interface, (&configuration).into())
+pub fn channel(
+    network_interface: &NetworkInterface,
+    configuration: Config,
+    #[cfg(feature = "psocket")]
+    socket: i32
```

Comment:
I think this should be part of the `Config` struct, rather than an additional argument.